demisto_sdk/commands/unify/yaml_unifier.py

```diff
@@ -0,0 +1,81 @@
+import io
+import os
+import sys
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from demisto_sdk.commands.common.constants import (DIR_TO_PREFIX,
+                                                   INTEGRATIONS_DIR,
+                                                   SCRIPTS_DIR)
+from demisto_sdk.commands.common.errors import Errors
+from demisto_sdk.commands.common.handlers import YAML_Handler
+from demisto_sdk.commands.common.tools import get_yml_paths_in_dir, print_error
+
+UNSUPPORTED_INPUT_ERR_MSG = 'Unsupported input. Please provide either: ' \
+                            '1. a directory of an integration or a script. ' \
+                            '2. a path of a GenericModule file.'
+
+
+class YAMLUnifier(ABC):
+    """Interface to YAML objects that need to be unified
+
+    Attributes:
+        package_path (str): The directory path to the files to unify.
+        dest_path (str, optional): The output dir to write the unified YAML to.
+        use_force(bool): Forcefully overwrites the preexisting yml if one exists.
+        yaml(YAML_Handler): Wrapper object to handle YAML files.
+        yml_path(str): The YAML file path.
+        yml_data(dict): The YAML doucment Python object.
+    """
+
+    def __init__(
+        self,
+        input: str,
+        output: Optional[str] = None,
+        force: bool = False,
+    ):
+        directory_name = ''
+        # Changing relative path to current abspath fixed problem with default output file name.
+        if input == '.':
+            input = os.path.abspath(input)
+        if not os.path.isdir(input):
+            print_error(UNSUPPORTED_INPUT_ERR_MSG)
+            sys.exit(1)
+        for optional_dir_name in DIR_TO_PREFIX:
+            if optional_dir_name in input:
+                directory_name = optional_dir_name
+
+        if not directory_name:
+            print_error(UNSUPPORTED_INPUT_ERR_MSG)
+
+        self.package_path = input
+        if self.package_path.endswith(os.sep):
+            self.package_path = self.package_path.rstrip(os.sep)
```

Comment:
The condition is redundant
```suggestion
        self.package_path = self.package_path.rstrip(os.sep)
```